app/src/main/java/com/mybus/service/MyBusService.java

```diff
@@ -0,0 +1,46 @@
+package com.mybus.service;
+
+import com.google.android.gms.maps.model.LatLng;
+import com.mybus.builder.MyBusServiceUrlBuilder;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+public class MyBusService extends GenericService {
+
+    /**
+     * @param origin
+     * @param destiny
+     * @return
+     */
+    public JSONObject searchRoutes(LatLng origin, LatLng destiny) throws IOException, JSONException {
+        //Create request to get a list of possible routes between two locations.
+        String url = MyBusServiceUrlBuilder.buildNexusUrl(origin.latitude, origin.longitude, destiny.latitude, destiny.longitude);
+        return new JSONObject(executeUrl(url));
+    }
+
+
+    /**
+     * @param mType
+     * @param mIdLine
+     * @param mDirection
+     * @param mStop1
+     * @param mStop2
+     * @param mIdLine2
+     * @param mDirection2
+     * @param mStop2L2
+     * @param mStop1L2
+     * @return
+     */
+    public JSONObject searchRoads(int mType, String mIdLine, String mDirection, String mStop1, String mStop2, String mIdLine2, String mDirection2, String mStop2L2, String mStop1L2) throws IOException, JSONException {
```

Comment:
The Signature should be:

```java
public RoadResult searchRoads(RoadSearch roadSearch)...
```

The parsing from JSON to model instances should be resolved inside the service.

And then create the RoadSearch class in order to encapsulate the parameters:

For more details:  http://refactoring.com/catalog/introduceParameterObject.html

```java
public class RoadSearch {

  private int mType;
  private String mIdLine;
  private String mDirection;
  ......

  //Constructor
  //Setters and getters
```
This makes the code more readable and understandable